flash/video/classification/data.py

```diff
@@ -0,0 +1,210 @@
+        return self.current_transform(sample)
+
+    def per_batch_transform_on_device(self, sample: _PYTORCHVIDEO_DATA) -> _PYTORCHVIDEO_DATA:
+        return self.current_transform(sample)
+
+
+class VideoClassificationData(DataModule):
+    """Data module for Video classification tasks."""
+
+    preprocess_cls = VideoClassificationPreprocess
+
+    @classmethod
+    def instantiate_preprocess(
+        cls,
+        clip_sampler: 'ClipSampler',
+        video_sampler: Type[Sampler],
+        decode_audio: bool,
+        decoder: str,
+        train_transform: Optional[Dict[str, Callable]],
+        val_transform: Optional[Dict[str, Callable]],
+        test_transform: Optional[Dict[str, Callable]],
+        predict_transform: Optional[Dict[str, Callable]],
+        preprocess_cls: Type[Preprocess] = None
+    ) -> Preprocess:
+        """
+        """
+        preprocess_cls = preprocess_cls or cls.preprocess_cls
+        preprocess: Preprocess = preprocess_cls(
+            clip_sampler, video_sampler, decode_audio, decoder, train_transform, val_transform, test_transform,
+            predict_transform
+        )
+        return preprocess
+
+    @classmethod
+    def from_paths(
+        cls,
+        train_folder: Optional[Union[str, pathlib.Path]] = None,
+        val_folder: Optional[Union[str, pathlib.Path]] = None,
+        test_folder: Optional[Union[str, pathlib.Path]] = None,
+        predict_folder: Union[str, pathlib.Path] = None,
+        clip_sampler: Union[str, 'ClipSampler'] = "random",
+        clip_duration: float = 2,
+        video_sampler: Type[Sampler] = RandomSampler,
+        clip_sampler_kwargs: Dict[str, Any] = None,
+        decode_audio: bool = True,
+        decoder: str = "pyav",
+        train_transform: Optional[Dict[str, Callable]] = None,
+        val_transform: Optional[Dict[str, Callable]] = None,
+        test_transform: Optional[Dict[str, Callable]] = None,
+        predict_transform: Optional[Dict[str, Callable]] = None,
+        batch_size: int = 4,
+        num_workers: Optional[int] = None,
+        preprocess_cls: Optional[Type[Preprocess]] = None,
+        **kwargs,
+    ) -> 'DataModule':
+        """
+
+        Creates a VideoClassificationData object from folders of videos arranged in this way: ::
+
+            train/class_x/xxx.ext
+            train/class_x/xxy.ext
+            train/class_x/xxz.ext
+            train/class_y/123.ext
+            train/class_y/nsdf3.ext
+            train/class_y/asd932_.ext
+
+        Args:
+            train_folder: Path to training folder. Default: None.
+            val_folder: Path to validation folder. Default: None.
+            test_folder: Path to test folder. Default: None.
+            predict_folder: Path to predict folder. Default: None.
+            clip_sampler: ClipSampler to be used on videos.
+            train_transform: Dictionnary of Video Clip transform to use for training set.
```

Comment:
Some missing docstrings for: clip_duration, video_sampler, clip_sampler_kwargs, decode_audio, and decoder

Reply:
Done !